test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+					testlog.Infof("cpus used by ovs services are: %s", maskSet.String())
+					Expect(cpus).To(Equal(maskSet.String()), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", cpus, maskSet.String())
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+				deleteTestPod(testpod)
+			})
 
-			By("checking the cpuset has balancing disabled")
-			cmd = []string{"cat", "/rootfs/sys/fs/cgroup/cpuset/ovs.slice/cpuset.sched_load_balance"}
-			ovsLoadBalancing, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(ovsLoadBalancing).To(Equal("0"))
+			It("[test_id:64102] Create and remove gu pods to verify affinity of ovs are changed appropriately", func() {
+				cpuID := onlineCPUSet.UnsortedList()[0]
+				smtLevel := nodes.GetSMTLevel(cpuID, workerRTNode)
+				if smtLevel < 2 {
+					Skip(fmt.Sprintf("designated worker node %q has SMT level %d - minimum required 2", workerRTNode.Name, smtLevel))
+				}
+				var testpod1, testpod2 *corev1.Pod
+				var err error
+
+				// Create testpod1
+				testpod1 = pods.GetTestPod()
+				testpod1.Namespace = testutils.NamespaceTesting
+				testpod1.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+
+				testpod1.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod1)
+				Expect(err).ToNot(HaveOccurred())
+				testpod1, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod1), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod1.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				tasksetcmd := []string{"taskset", "-pc", "1"}
+				testpod1Cpus, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod1, tasksetcmd)
+				testlog.Infof("%v pod is using %v cpus", testpod1.Name, string(testpod1Cpus))
+
+				// Create testpod1
+				testpod2 = pods.GetTestPod()
+				testpod2.Namespace = testutils.NamespaceTesting
+				testpod2.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+				testpod2.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod2)
+				Expect(err).ToNot(HaveOccurred())
+				testpod2, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod2), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod1.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				By("Getting the container cpuset.cpus cgroup")
```

Comment:
This code for getting the Pod cpuset is repeated multiple times in the file, could you maybe extract it to a function?

Reply:
fixed in the latest commit.